app/controllers/internal/broadcasts_controller.rb

```diff
@@ -28,22 +17,39 @@ def edit
     @broadcast = Broadcast.find_by!(id: params[:id])
   end
 
-  def index
-    @broadcasts = if params[:type_of]
-                    Broadcast.where(type_of: params[:type_of].capitalize)
-                  else
-                    Broadcast.all
-                  end.order(title: :asc)
+  def create
+    @broadcast = Broadcast.new(broadcast_params)
+
+    if @broadcast.save
+      flash[:success] = "Broadcast has been created!"
+      redirect_to internal_broadcasts_path
+    else
+      flash[:danger] = @broadcast.errors.full_messages.to_sentence
+      render new_internal_broadcast_path
+    end
+  end
+
+  def update
+    @broadcast = Broadcast.find_by!(id: params[:id])
+
+    if @broadcast.update(broadcast_params)
+      flash[:success] = "Broadcast has been updated!"
+      redirect_to internal_broadcasts_path
+    else
+      flash[:danger] = @broadcast.errors.full_messages.to_sentence
+      render :edit
+    end
   end
 
   def destroy
-    broadcast = Broadcast.find_by!(id: params[:id])
-    if broadcast.destroy
+    @broadcast = Broadcast.find_by!(id: params[:id])
```

Comment:
A nitpick 😬  - in the future, I believe you can use `find(params[:id])`

Reply:
ohhhh duh of course. fixing now. can you tell i did this in like 2 minutes? 😹